src/main/java/com/nutomic/syncthingandroid/fragments/SettingsFragment.java

```diff
@@ -392,6 +427,13 @@ public void onClick(DialogInterface dialogInterface, int i) {
                         })
                         .show();
                 return true;
+            case SyncthingService.PREF_SYNC_ONLY_WIFI_SSIDS:
+                WifiManager wifiManager = (WifiManager) getActivity().getSystemService(Context.WIFI_SERVICE);
+                if (wifiManager == null || wifiManager.getConfiguredNetworks() == null) {
+                    mSyncOnlyOnSSIDs.getDialog().dismiss();
```

Comment:
It does, the dialog is actually shown before the click handler is called, so this is necessary to close the dialog to only show the Toast.